### PR TITLE
Add more robust OH tests for find method

### PR DESCRIPTION
**Pull request by Amir-61**
Add more robust OH tests for find() method

Related to https://github.com/strongloop/loopback-datasource-juggler/pull/1149#issuecomment-256047743

/to: @superkhau @loay 
/cc: @bajtos

### CODE REVIEW

File: test/persistence-hooks.suite.js [hunk 1]
```diff
@@ -169,6 +190,21 @@ module.exports = function(dataSource, should, connectorCapabilities) {
         });
       });
 
+      it('applies updates one specific instance from `loaded` hook when near filter is not used',
```
Comment (superkhau) [inline test/persistence-hooks.suite.js]: applies updates to one

File: test/persistence-hooks.suite.js [hunk 2]
```diff
@@ -109,6 +109,23 @@ module.exports = function(dataSource, should, connectorCapabilities) {
           });
       });
 
+      it.only('triggers correct hooks when we have multiple instances', function(done) {
```
Comment (superkhau) [inline test/persistence-hooks.suite.js]: triggers the correct hooks when multiple instances exist

File: test/persistence-hooks.suite.js [hunk 3]
```diff
@@ -109,6 +109,23 @@ module.exports = function(dataSource, should, connectorCapabilities) {
...
+        monitorHookExecution();
+
+        TestModel.find(
+          function(err, list) {
```
Comment (superkhau) [inline test/persistence-hooks.suite.js]: move this up to the same line as find

File: test/persistence-hooks.suite.js [hunk 4]
```diff
@@ -109,6 +109,23 @@ module.exports = function(dataSource, should, connectorCapabilities) {
...
+            if (err) return done(err);
+
+            // Please note `loaded` hook gets trigered for each item (2 items) of the result set
```
Comment (superkhau) [inline test/persistence-hooks.suite.js]: remove comment, this is expected from the wording of the test

File: test/persistence-hooks.suite.js [hunk 5]
```diff
@@ -119,7 +136,8 @@ module.exports = function(dataSource, should, connectorCapabilities) {
           });
       });
 
-      it('triggers correct hooks when near filter is used', function(done) {
+      it('triggers the loaded hook multiple times when multiple instances exist with near filter',
```
Comment (superkhau) [inline test/persistence-hooks.suite.js]: "when near filter is given" or "when near filter is used" to match the terms used in other test cases

**Comment by superkhau** [comment]
@slnode test please

File: test/persistence-hooks.suite.js [hunk 6]
```diff
@@ -80,8 +80,10 @@ module.exports = function(dataSource, should, connectorCapabilities) {
           TestModel.create({name: 'second'}, function(err) {
             if (err) return done(err);
             var location1 = new GeoPoint({lat: 10.2, lng: 6.7});
+            var location1 = new GeoPoint({lat: 10.3, lng: 6.8});
```
Comment (bajtos) [inline test/persistence-hooks.suite.js]: `location2`? `location1` is already in use.

Comment (Amir-61) [inline test/persistence-hooks.suite.js]: It is fixed last night by https://github.com/strongloop/loopback-datasource-juggler/pull/1160

**Pull request merged**